tests/whisper_protocol/crypto_wrappers/curve25519.rs

```diff
@@ -54,19 +59,74 @@ pub fn generate_private_key() -> PrivateKey{
     PrivateKey::from_bytes(*private_key)
 }
 
-#[derive(Clone)]
-#[derive(PartialEq)]
 pub struct PublicKey {
```

Comment:
consider making this a packed struct, with a DJB_TYPE byte, and a [u8;32] key.
i think it'd be cleaner than having an offset of 1.